src/Plugin/Field/FieldWidget/AvPortalWidget.php

```diff
@@ -55,37 +80,72 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
   /**
    * {@inheritdoc}
    */
-  public static function validate(array $element, FormStateInterface $form_state): void {
+  public static function validatePhoto(array $element, FormStateInterface $form_state, $validated_type): void {
+    self::validate($element, $form_state, self::AVPORTAL_PHOTO);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function validateVideo(array $element, FormStateInterface $form_state, $validated_type): void {
+    self::validate($element, $form_state, self::AVPORTAL_VIDEO);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function validate(array $element, FormStateInterface $form_state, $validated_type): void {
     $value = $element['value']['#value'];
 
-    $patterns = [
-      // url: http://ec.europa.eu/avservices/video/player.cfm.
-      '@ec\.europa\.eu/avservices/video/player\.cfm\?(.+)@i',
-      // url: http://ec.europa.eu/avservices/play.cfm.
-      '@ec\.europa\.eu/avservices/play\.cfm\?(.+)@i',
-    ];
+    $patterns = self::getPatterns();
 
-    foreach ($patterns as $pattern) {
-      if (preg_match($pattern, $value)) {
+    foreach ($patterns as $pattern => $type) {
+      if (preg_match($pattern, $value) && $type == $validated_type) {
         return;
       }
     }
 
     $form_state->setError($element['value'], t('Invalid URL format specified.'));
```

Comment:
It would be interesting (but not a requirement) if we notice that the url being used belongs to another source type (using a photo url in a video formatter) we let the user now with a custom message (e.g. "Invalid URL format specified. It appears that you are using a video url on a photograph field").